### ISSUE TITLE
Glossy can fail to converge for large docs and terms with multiple uses

### ISSUE THREAD

**Issue by Alex-Muirhead** [swaits/typst-collection#11]
When using Glossy with a large document (but not necessarily a large number of terms in the glossary), the document can fail to converge when
1. The `glossary` command is used to print the definitions **and**
2. Any term is used more than once. 
This can occur with a minimum of one term being used twice, even on the same page (i.e. `@LEO @LEO`). 

If all terms are only used once, the problem does not appear.

I haven't been able to create a trivial MVP to recreate this yet unfortunately, sorry. I only found this in a document with approx 30 pages of (non-trivial) content.

**Comment by swaits** [comment]
I've experienced this too. Glossy uses `context{}` a good bit to find all the term usages throughout your doc and then generate backlinks to them (for the index).

I've found in the past that I can get it to converge by adjusting other, more dynamic, parts of my document. For example, situations where Typst has to iterate to find a final layout (because specific components' layouts impacts others, recursively and forever).

Basically, I don't have a good solution to this now. I'm open to fixing it in Glossy. But, I've been bucketing it as a Typst issue thus far.

**Comment by embediver** [comment]
I just wanted to report, that I am facing this problem in a thesis I am writing.

I didn't use many dynamic elements in my document, but the layout doesn't converge.

Is there a possibility to just deactivate backlinks with a option or a special theme?
I assume that would help greatly.

**Comment by swaits** [comment]
In the latest version I'm disabling forward-links by default. Until this is resolved, I may also disable back-links by default as well.

I'll try to publish that this weekend.

**Comment by swaits** [comment]
FYI, the version with forward links off by default is now awaiting upstream merge. It'll be version 0.8.0.
